### ISSUE TITLE
Error while using Brave browser (potentially any other adblocker)

### ISSUE THREAD

**Issue by tregismoreira** [onfido/onfido-sdk-ui#312]
Hey guys,

We've been using Onfido SDK UI on our website and it's working pretty well. But sometimes we get a support request from users having troubles to send the documents. After a few tests and deep analysis, I think I've found a pattern: It doesn't support the Brave browser, and potentially any other adblocker extension.

Here are the 2 errors/bugs:

1. The picture is not taken automatically at all, then we have to upload the document manually;
2. When we select the document to upload, we get the error "Connection lost" (see the attachment).

We are suggesting our users to try another browser and it works.

I hope you are able to fix this bug soon ;)

![screenshot at feb 16 11-40-57](https://user-images.githubusercontent.com/197236/36310343-d580bf48-130e-11e8-8b41-9c0b44f7def7.png)

**Comment by 99darwin** [comment]
Any updates on this issue? Having the same issue on Chrome on macOS. 

Network errors returned: 
```
POST https://api.onfido.com/v2/documents 401 ()
POST https://sentry.io/api/109946/store/?sentry_version=7&sentry_client=raven-js%2F3.9.1&sentry_key=[key-omitted] 403 
```

**Comment by 99darwin** [comment]
I solved this issue by waiting until after the documents were uploaded to create the check. Hope this helps someone else!

**Comment by rfreitas** [comment]
Good news @99darwin.
I'm closing this ticket, even though I think the two problems might not be exactly the same.

**Comment by ah-idx9** [comment]
Is there a resolution to this ? We are facing the same issue with the onfido SDKs

**Comment by stefaniacardenas** [comment]
Hi @ah-idx9, can you describe your issue in more details? There are multiple issues described in the thread above so I want to understand which one you are facing. Which version of the SDK are you using? Which configuration parameters are you passing to the `init()` function? On which browsers/devices/OS? Can you describe the steps to reproduce the problem?

**Comment by ah-idx9** [comment]
Hello @stefaniacardenas 
We were able to resolve the issue. I will describe this here. 
I suspect you will see more cases of this in the near future.

We are using the onfido-ui-sdk (v 5.9.2) to allow customers to upload documents and kick off KYC checks.

A few weeks ago, we started experiencing issues with the kyc process (See the Connection Lost, please try again) image in the thread above. This is the exact issue we were experiencing. This was in code that has been working for months prior to when we started seeing the issue.

On closer inspection, the upload API being called by the onfido-ui-sdk was returning a 401, and the specific error was "Invalid referer". We use the  Generate SDK Token API (https://documentation.onfido.com/#generate-sdk-token) with a **referer** to instantiate the check process. 

Turns out, the referer being passed to onfido was different than the referer we were seeing on our API side. This started happening recently, so that explains the failures that started recently.

A bit more digging and we came across this :
https://developers.google.com/web/updates/2020/07/referrer-policy-new-chrome-default

With recent versions of chrome (and my guess is brave), the default referrer policy (this governs the referrer string that is passed to 3rd party sites) is  **_strict-origin-when-cross-origin_**. Before version 85 of chrome, this default policy was **_no-referrer-when-downgrade_** 

Since in our code, we were using the incoming referrer to create the sdk token, the sdk token had a referer with the complete path of the URL. e.g. 

**https://a.com/foo/bar?baz=bozo**

However, when chrome called a third party (such as onfido apis called by onfido-sdk-ui), the referrer being passed was just **https:/a.com/**. This is due to the new default referrer policy on chrome. 

This was causing the API call to fail with 401.

**The solution**
Our app backend is fronted by nginx, so we added a Referrer-Policy response header to return **_no-referrer-when-downgrade_** 

_**nginx.conf**_

http {
       ...
        add_header Referrer-Policy no-referrer-when-downgrade;
      ...
}

Basically configure your webserver to add this response header to requests. This should be possible in all web servers.

This has solved the issue. An improvement would be to set this header as **_strict-origin-when-cross-origin_** and update the code to pass only the origin as referer to the onfido sdk generate token. 

Next version.

HTH
